presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1044,24 +1048,47 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     }
 
     @SqlNullable
-    @Description("Returns an array of points in a linestring")
+    @Description("Returns an array of points in a geometry")
     @ScalarFunction("ST_Points")
     @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
     public static Block stPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
-        validateType("ST_Points", geometry, EnumSet.of(LINE_STRING));
+        Geometry geometry = JtsGeometrySerde.deserialize(input);
+        validateType("ST_Points", geometry, EnumSet.of(LINE_STRING, POLYGON, POINT, MULTI_POINT,
```

Comment:
We generally place arguments either all on the same line or all in separate lines. In this case, I'd put all the enum values on the same line. 

Also, let's make it a `private static final` constant to avoid creating the enum set in every call.

Reply:
Thanks @martint for your review. I have made the changes based on your comments.